File: ClientPhoneBook/Models/ContactModel.cs
using System.ComponentModel.DataAnnotations;

namespace ClientPhoneBook.Models
{
    public class ContactModel
    {
        [Key]
        public int ContactId { get; set; }

        [Required]
        [StringLength(15)]
        public string FirstName { get; set; }

        [Required]
        [StringLength(15)]
        public string LastName { get; set; }

        [Required]
        [StringLength(50)]
        public string Email { get; set; }

        [Required]
        [StringLength(15)]
        public string Phone { get; set; }

        [Required]
        [StringLength(15)]
        public string Company { get; set; }
    }
}

File: PhoneBookApp/src/app/models/contact.model.ts
import { ContactCountry } from "./contact.country.model";
import { ContactState } from "./contact.state.model";

export interface Contact{
    contactId:number;
    countryId:number;
    stateId:number;
    firstName:string;
    lastName:string;
    email:string;
    phone:string;
    image:string;
    country:ContactCountry;
    state:ContactState;
    imageByte:string;
    birthDate:Date|null;
    company:string;
    gender:string;
    favourites:boolean;

}